src/pages/Dashboard/data.ts

```diff
@@ -1,137 +1,57 @@
-    id: 9999999,
-    name: "Djibouti",
-    continent: "Afrika",
-    last_modified: "2023-05-14T12:47:24.455403",
-    total: 2780,
-    lgbt: 200,
-    status: {
-      Accepted: 1000,
-      Rejected: 1780,
-      Unknown: 100,
-    },
-    gender: {
-      Male: 60,
-      Female: 10,
-      Other: 65,
-      Unknown: 1,
-    },
-  },
-  {
-    id: 99999999,
-    name: "Ecuador",
-    continent: "Sydamerika",
-    last_modified: "2023-05-14T12:47:24.455403",
-    total: 4800,
-    lgbt: 400,
-    status: {
-      Accepted: 2600,
-      Rejected: 2200,
-      Unknown: 100,
-    },
-    gender: {
-      Male: 30,
-      Female: 40,
-      Other: 170,
-      Unknown: 5,
-    },
-  },
-  {
-    id: 999999999,
-    name: "Finland",
-    continent: "Europa",
-    last_modified: "2023-05-14T12:47:24.455403",
-    total: 500,
-    lgbt: 400,
-    status: {
-      Accepted: 90,
-      Rejected: 410,
-      Unknown: 100,
-    },
-    gender: {
-      Male: 50,
-      Female: 50,
-      Other: 25,
-      Unknown: 10,
-    },
-  },
-  {
-    id: 9999999999,
-    name: "Ghana",
-    continent: "Afrika",
-    last_modified: "2023-05-14T12:47:24.455403",
-    total: 3490,
-    lgbt: 310,
-    status: {
-      Accepted: 1300,
-      Rejected: 2190,
-      Unknown: 100,
-    },
-    gender: {
-      Male: 90,
-      Female: 120,
-      Other: 25,
-      Unknown: 5,
-    },
-  },
-];
+import { useEffect, useState } from 'react';
+
+// Definition of country object type
+interface Country {
+  id: number;
+  name: string;
+  continent: string;
+  last_modified: string;
+  total: number;
+  lgbt: number;
+  status: {
+    Accepted: number;
+    Rejected: number;
+    Unknown: number;
+  };
+  gender: {
+    Male: number;
+    Female: number;
+    Other: number;
+    Unknown: number;
+  };
+}
+
+// This function performs the HTTP request that gets all the countries from the database.
+// Each country in the returned lists is of the above format
+export const countryRequest = () => {
+  const [countries, setCountries] = useState<Country[]>([]);
+
+  // HTTP request that gets a list of all country entries from the database
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch("/api/countries");
+        const data = await response.json();
+
+        setCountries(data);
+      } catch (error) {
+        // eslint-disable-next-line
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  // Copy of list to avoid messing with "countries"
+  const fetchedCountries = countries;
+
+  // We sort the countries based on number of cases in decreasing order
+  fetchedCountries.sort((a, b) => b.total - a.total);
+
+  // Select the first 7 of the sorted countries as the default data
+  const defaultCountries = fetchedCountries.slice(0, 7);
+
+  // Return list of all countries, together with a list of default countries to display
+  return [countries, defaultCountries];
+}
```

Comment:
and then maybe we dont need this file anymore? if you want to keep the typing here and delete the other instances then maybe rename it to types.ts or something better than data.ts since it isnt holding data anymore